app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -110,9 +110,18 @@ def sync
     transaction do
       current = configuration_script_payloads.index_by(&:name)
 
-      playbooks_in_git_repository.each do |f|
-        found = current.delete(f) || self.class.module_parent::Playbook.new(:configuration_script_source_id => id)
-        found.update!(:name => f, :manager_id => manager_id)
+      git_repository.update_repo
+      git_repository.with_worktree do |worktree|
+        worktree.ref = scm_branch
+        worktree.blob_list.each do |filename|
+          next unless playbook_dir?(filename)
+
+          content = worktree.read_file(filename)
+          next unless playbook?(filename, content)
```

Comment:
If the content is an $ANSIBLE_VAULT, it's a playbook, but we can't read it because it's encrypted. (That's the `return true` check in `playbook?`.  What will happen in that case?

Reply:
I think it will just write the encrypted data into the table, but not sure we want that?

Reply:
Oh interesting I didn't know that's what the `$ANSIBLE_VAULT` check was.  Yes it would write the encrypted payload to the record, I can pull out an `encrypted_playbook?` method and skip if it is encrypted.

Reply:
It's up to you - not sure what the best approach is

Reply:
FYI - https://docs.ansible.com/ansible/latest/vault_guide/vault_using_encrypted_content.html#format-of-files-encrypted-with-ansible-vault